source/common/network/transport_socket_options_impl.h

```diff
@@ -45,6 +52,16 @@ class TransportSocketOptionsUtility {
    */
   static TransportSocketOptionsSharedPtr
   fromFilterState(const StreamInfo::FilterState& stream_info);
+  /**
+   * Construct TransportSocketOptions from StreamInfo::FilterState, using UpstreamServerName
+   * and ApplicationProtocols key in the filter state as well as optional PROXY protocol header
+   *
+   * @returns TransportSocketOptionsSharedPtr a shared pointer to the transport socket options,
+   * nullptr if nothing is in the filter state or no PROXY protocol info is supplied.
+   */
+  static TransportSocketOptionsSharedPtr fromFilterStateWithProxyProtocolHeader(
```

Comment:
Is this function needed? Couldn't the proxy protocol data be part of the `FilterState`?

Reply:
I'll double check this 👍

Reply:
Yeah, definitely can just add the proxy proto data to the `FilterState`.